src/types.ts

```diff
@@ -1127,18 +1145,20 @@ export class Signature {
   }
 
   /** Creates a clone of this signature that is safe to modify. */
-  clone(): Signature {
+  clone(requiredParameters: i32 = this.requiredParameters, hasRest: bool = this.hasRest): Signature {
     let parameterTypes = this.parameterTypes;
     let numParameterTypes = parameterTypes.length;
     let cloneParameterTypes = new Array<Type>(numParameterTypes);
     for (let i = 0; i < numParameterTypes; ++i) {
       unchecked(cloneParameterTypes[i] = parameterTypes[i]);
     }
-    return new Signature(
+    return Signature.new(
```

Comment:
Is this actually going to create a clone?
```suggestion
    return Signature.create(
```